common/utils/src/test/scala/org/apache/spark/util/StructuredLoggingSuite.scala

```diff
@@ -56,7 +56,7 @@ trait LoggingSuiteBase
   def basicMsgWithEscapeChar: String = "This is a log message\nThis is a new line \t other msg"
 
   def basicMsgWithEscapeCharMDC: LogEntry =
-    log"This is a log message\nThis is a new line \t other msg"
+    log"This is a log message" + LF + log"This is a new line " + TAB + log" other msg"
```

Comment:
Although the above writing may seem `a bit ugly`, so far, the only method I can find that meets the `requirements` and is relatively acceptable